x/tally/keeper/endblock.go

```diff
@@ -247,7 +248,7 @@ func (k Keeper) FilterAndTally(ctx sdk.Context, req types.Request) (TallyResult,
 	if err != nil {
 		return result, k.logErrAndRet(ctx, err, types.ErrGettingMaxTallyGasLimit, req)
 	}
-	gasLimit := min(req.TallyGasLimit, maxGasLimit)
+	gasLimit := min(req.TallyGasLimit, maxGasLimit) - filterResult.GasUsed
```

Comment:
We should verify that the `filterResult.GasUsed` is larger than the `gasLimit` to prevent insanely high gas limits :p

Reply:
changed it to 
```
gasLimit := max(min(req.TallyGasLimit, maxGasLimit)-filterResult.GasUsed, 0)
```

Reply:
That won't do anything since they're unsigned integers right? 0 - 100 = 18446744073709551516 in uint64 math.

Reply:
duh.. thanks again. fixed it to
```
	var gasLimit uint64
	if min(req.TallyGasLimit, maxGasLimit) > filterResult.GasUsed {
		gasLimit = min(req.TallyGasLimit, maxGasLimit) - filterResult.GasUsed
	} else {
		gasLimit = 0
	}
```